core/domain/topic_services.py

```diff
@@ -984,9 +984,13 @@ def assign_role(committer, assignee, new_role, topic_id):
         old_role = topic_domain.ROLE_MANAGER
 
     if new_role == topic_domain.ROLE_MANAGER:
-        if topic_rights.is_manager(assignee.user_id):
-            raise Exception('This user already is a manager for this topic')
-        topic_rights.manager_ids.append(assignee.user_id)
+        # An exception shouldn't be raised if the user is already a manager of
```

Comment:
But if the user's topic manager role has been removed and then reassigned to the same topic, the user wouldn't be "already a manager of the topic", right? So ... I'm not sure I understand the logic here.

Reply:
Yeah, that was the issue. When the role was removed, the topic rights object we not modified, only the role itself was changed. So, when the user was assigned back to the same topic, the error was raised.

Reply:
Oh I see. But shouldn't the correct fix then be to modify the topic rights object when the role is removed?

Reply:
Yeah, done. So, now when a user is assigned any role and their previous role was topic manager, all topicsrights objects with that user as an  assignee is modified to remove that user.